### ISSUE TITLE
False positive "not-callable" after `is not None`

### ISSUE THREAD

**Issue by jab** [pylint-dev/pylint#6005]
### Bug description

Minimized reproducer:
```python
# pylint: disable=too-few-public-methods

"Module to repro false positive 'not-callable' error."""

import weakref


class Repro:
    """Class to repro false positive bug."""

    def __init__(self):
        self._weak = None

    def repro(self):
        """Method to repro false positive bug."""
        weak = getattr(self, '_weak', None)
        if weak is not None:
            strong = weak()  # false positive error here
            if strong is not None:
                return strong
        self._weak = weakref.ref(self)
        return self._weak()
```

See the following line for a real-world example where this false positive is occurring: https://github.com/jab/bidict/blob/caf703e959ed4471bc391a7794411864c1d6ab9d/bidict/_base.py#L185

### Command used

```shell
pylint repro.py
```

### Pylint output

```shell
************* Module repro
repro.py:18:21: E1102: weak is not callable (not-callable)
```

### Expected behavior

No error should be flagged here.

### Pylint version

```shell
pylint 2.13.2
astroid 2.11.2
Python 3.10.2 (main, Feb  2 2022, 07:36:01) [Clang 12.0.0 (clang-1200.0.32.29)]
```

**Comment by jacobtylerwalls** [comment]
It's possible this is addressed by https://github.com/PyCQA/astroid/pull/1189

**Comment by Pierre-Sassoulas** [comment]
Closing in favor of #7412, more generic and detailed.
